qiskit_experiments/library/randomized_benchmarking/mirror_rb_experiment.py

```diff
@@ -0,0 +1,428 @@
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Mirror RB Experiment class.
+"""
+import warnings
+from typing import Union, Iterable, Optional, List, Sequence, Tuple
+from numbers import Integral
+import itertools
+import numpy as np
+from numpy.random import Generator, BitGenerator, SeedSequence
+
+from qiskit.circuit import QuantumCircuit, Instruction, Barrier
+from qiskit.quantum_info.operators import Pauli
+from qiskit.providers.backend import Backend
+from qiskit.providers.options import Options
+from qiskit.exceptions import QiskitError
+from qiskit.circuit.library import CXGate, CYGate, CZGate, ECRGate, SwapGate, iSwapGate
+
+from .rb_experiment import StandardRB, SequenceElementType
+from .mirror_rb_analysis import MirrorRBAnalysis
+from .clifford_utils import (
+    compute_target_bitstring,
+    inverse_1q,
+    _clifford_1q_int_to_instruction,
+)
+from .sampling_utils import (
+    BaseSampler,
+    EdgeGrabSampler,
+    SingleQubitSampler,
+    GateTypeT,
+    GateInstruction,
+)
+
+# two qubit gates that are their own inverse
+_self_adjoint_gates = [CXGate, CYGate, CZGate, ECRGate, SwapGate, iSwapGate]
+
+
+class MirrorRB(StandardRB):
+    """An experiment to measure gate infidelity using random mirrored layers of Clifford
+    and Pauli gates.
+
+    # section: overview
+        Mirror randomized benchmarking (mirror RB) estimates the average error rate of
+        quantum gates using gate layers sampled from a distribution that are then
+        inverted in the second half of the circuit.
+
+        The default mirror RB experiment generates circuits of layers of Cliffords,
+        consisting of single-qubit Cliffords and a two-qubit gate such as CX,
+        interleaved with layers of Pauli gates and capped at the start and end by a
+        layer of single-qubit Cliffords. The second half of the Clifford layers are the
+        inverses of the first half of Clifford layers. After running the circuits on a
+        backend, various quantities (success probability, adjusted success probability,
+        and effective polarization) are computed and used to fit an exponential decay
+        curve and calculate the EPC (error per Clifford, also referred to as the average
+        gate infidelity) and entanglement infidelity (see references for more info).
+
+    # section: analysis_ref
+        :class:`MirrorRBAnalysis`
+
+    # section: manual
+        :doc:`/manuals/verification/mirror_rb`
+
+    # section: reference
+        .. ref_arxiv:: 1 2112.09853
+        .. ref_arxiv:: 2 2008.11294
+        .. ref_arxiv:: 3 2204.07568
+
+    """
+
+    def __init__(
+        self,
+        physical_qubits: Sequence[int],
+        lengths: Iterable[int],
+        distribution: BaseSampler = EdgeGrabSampler,
+        start_end_clifford: bool = True,
```

Comment:
Yeah, I think it's better to use `**experiment_options` to avoid writing the same docstring and arguments twice. But let's discuss this later in the context of the other experiments.

Reply:
Fair enough. Let's do in follow-up.